### PR TITLE
Try to remove ARIA cipher

### PR DESCRIPTION
**Pull request by InfoHunter**
There are a bundle of cryptographic algorithms that are useless in BabaSSL's aimed scenarios. This is the first try that removes one of them - the ARIA cipher. This will reduce the size of BabaSSL's code base.

Only one thing we need to be carefully: are there going-to-be-removed algorithms still useful for other components inside an OS distribution, like Anolis OS for instance? So could you check if the removal of ARIA will be no harm to the AnolisOS requirements, @hustliyilin ?

### CODE REVIEW

**Comment by uudiin** [comment]
It is no problem to remove ARIA algorithm. We will do some tests for Anolis. If there is a compatibility issue, we will add patch to the RPM to solve it.

**Comment by uudiin** [comment]
Another question is: why not consider compiling to control the size of the library, for example, in some scenarios, ARIA or even more algorithms are not compiled by default.

**Comment by InfoHunter** [comment]
> Another question is: why not consider compiling to control the size of the library, for example, in some scenarios, ARIA or even more algorithms are not compiled by default.

Consider this situation: if some bugs are found in the code that we don't use, even the code we don't compile, we still need to check if the bug affects the whole library or not; at least we need to explain this to the users. So if we remove more useless code, the effort we need to deal with this kind of thinks would also be reduced as well.

**Comment by hustliyilin** [comment]
@uudiin Could you help to evaluate this pr together for AnolisOS, thanks.

**Review by uudiin — APPROVED** [review]

File: crypto/objects/obj_dat.h [hunk 1]
```diff
@@ -2152,153 +2131,153 @@ static const ASN1_OBJECT nid_objs[NUM_NID] = {
...
-    {"id-rsassa-pkcs1-v1_5-with-sha3-256", "RSA-SHA3-256", NID_RSA_SHA3_256, 9, &so[7156]},
-    {"id-rsassa-pkcs1-v1_5-with-sha3-384", "RSA-SHA3-384", NID_RSA_SHA3_384, 9, &so[7165]},
-    {"id-rsassa-pkcs1-v1_5-with-sha3-512", "RSA-SHA3-512", NID_RSA_SHA3_512, 9, &so[7174]},
-    {"ARIA-128-CCM", "aria-128-ccm", NID_aria_128_ccm, 9, &so[7183]},
-    {"ARIA-192-CCM", "aria-192-ccm", NID_aria_192_ccm, 9, &so[7192]},
-    {"ARIA-256-CCM", "aria-256-ccm", NID_aria_256_ccm, 9, &so[7201]},
-    {"ARIA-128-GCM", "aria-128-gcm", NID_aria_128_gcm, 9, &so[7210]},
-    {"ARIA-192-GCM", "aria-192-gcm", NID_aria_192_gcm, 9, &so[7219]},
-    {"ARIA-256-GCM", "aria-256-gcm", NID_aria_256_gcm, 9, &so[7228]},
+    { NULL, NULL, NID_undef },
```
Comment (dongbeiouba) [inline crypto/objects/obj_dat.h:2134]: Why so many NID_undef?

Comment (hustliyilin) [reply crypto/objects/obj_dat.h:2134]: > 

Each object is assigned a fixed NID_num. for example, `sshkdf` is 1203. If this value is changed, the dependent application (ssh) will be incompatible. so if you want to delete a previous NID, you must add a NID_undef

**Comment by dongbeiouba** [comment]
是不是忘记更新crypto/objects/obj_mac.num 文件了？

**Comment by InfoHunter** [comment]
Fixed

**Review by dongbeiouba — APPROVED** [review]

**Pull request merged**